### PR TITLE
Test picking

### LINKED ISSUE #3265: heatmap and pick
`pick`ing on a heatmap always returns 0 no matter what square was clicked on:

```
julia> fig, ax, hm = heatmap(rand(3,3))
```
Screenshot 2023-09-28 at 9 43 56 AM

```
julia> on(events(fig).mousebutton, priority = 2) do event
           if event.button == Mouse.left && event.action == Mouse.press
               plt, i = pick(fig)
               @info plt, i
           end
           return Consume(false)
       end
ObserverFunction defined at REPL[41]:2 operating on Observable(MouseButtonEvent(none, release))

julia> [ Info: (nothing, 0)
[ Info: (Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}}, 0)
[ Info: (Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}}, 0)
[ Info: (Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}}, 0)
[ Info: (Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}}, 0)
[ Info: (Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}}, 0)
[ Info: (Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}}, 0)
[ Info: (Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}}, 0)
```

would be super useful if this returned a cartesian or linear index, much like it does for scatter:

```
julia> fig, ax, li = scatter([1,2,3],[1,2,3])

julia> on(events(fig).mousebutton, priority = 2) do event
           if event.button == Mouse.left && event.action == Mouse.press
               plt, i = pick(fig)
               @info plt, i
           end
           return Consume(false)
       end
ObserverFunction defined at REPL[43]:2 operating on Observable(MouseButtonEvent(none, release))

julia> [ Info: (Scatter{Tuple{Vector{Point{2, Float32}}}}, 3)
[ Info: (Scatter{Tuple{Vector{Point{2, Float32}}}}, 2)
[ Info: (Scatter{Tuple{Vector{Point{2, Float32}}}}, 1)
```

```
julia> VERSION
v"1.9.3"

julia> Sys.MACHINE
"arm64-apple-darwin22.4.0"

  [e9467ef8] GLMakie v0.8.10
```

### PR DESCRIPTION
**Pull request by ffreyer**
# Description

This will add picking tests for all primitives in GLMakie and WGLMakie and probably fix a few issues on the side.
TODO:
- [x] GLMakie picking tests
- [x] WGLMakie picking tests
- [x] make sure heatmap actually tests a heatmap path
- [x] consider reworking image, heatmap and surface so that the index relates to the given matrix
  - [x] GLMakie
  - [x] WGLMakie

Other fixes/Changes:
- fixed edge case of voxel picking indices going out of bounds
- changed strict equality in clip planes modelinv check to isapprox, matching GLMakie
- fixed arg based voxel placement & scaling in WGLMakie
- DataInspector now relies on heatmap picking index instead of manually calculating it

closes #3265

## Type of change

- Bug fix: Voxel indices
- New feature: useful image, heatmap & surface picking indices
- testing: picking tests for GLMakie & WGLMakie

## Checklist

- [x] Added an entry in CHANGELOG.md (for new features and breaking changes)
- [x] Added or changed relevant sections in the documentation
- [x] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

File: WGLMakie/test/picking.jl [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    i = image!(scene, 80..110, 20..50, rand(RGBf, 2, 2))
+    s = surface!(scene, 80..110, 80..110, rand(2, 2))
+    hm = heatmap!(scene, [80, 110], [140, 170], [1 2; 3 4])
+    # mesh coloring should match triangle placements
+    m = mesh!(scene, Point2f.([80, 80, 110, 110], [200, 230, 200, 230]), [1 2 3; 2 3 4], color = [1,1,1,2])
+    vx = voxels!(scene, [65, 125], [245, 305], [-1, 1], reshape([1,2,3,4], (2,2,1)), shading = NoShading)
+    vol = volume!(scene, 80..110, 320..350, -1..1, rand(2,2,2))
+
+    # render one frame to generate picking texture
+    colorbuffer(scene);
```
Comment (EdsterG) [inline WGLMakie/test/picking.jl:23]: Ah, so this was the trick for rendering a frame in CI.

Comment (ffreyer) [reply WGLMakie/test/picking.jl:23]: It might also depend on where the file is included. I (now) have it after the refimg tests since those need to work the same for every backend, and it seems to work

**Comment by ffreyer** [comment]
Test failures are caused by `scatter(f[2, 1], -1..1, x -> x^2)` calls generating a different distribution of points. I don't know what exactly is causing this, but between master and this pr the julia version updated from 1.10.5 to 1.11.0 and a bunch of base packages got Versions now. I assume something changed there to cause this
